json/src/main/kotlin/ktx/json/serializer.kt

```diff
@@ -0,0 +1,42 @@
+package ktx.json
+
+import com.badlogic.gdx.utils.Json
+import com.badlogic.gdx.utils.JsonValue
+
+/**
+ * Wrapping interface around [com.badlogic.gdx.utils.Json.Serializer].
+ * Provides null-safety for the methods.
+ */
+interface KtxSerializer<T> : Json.Serializer<T> {
+  override fun write(json: Json, obj: T, knownType: Class<*>?)
+
+  override fun read(json: Json, jsonData: JsonValue, type: Class<*>?): T
+}
+
+/**
+ * Wrapping interface around [com.badlogic.gdx.utils.Json.Serializer].
+ * Provides null-safety and convenient interface for serializer
+ * that is only able to [read]. The [write] method is no-op.
+ */
+interface KtxReadOnlySerializer<T> : Json.Serializer<T> {
+  override fun write(json: Json, obj: T, knownType: Class<*>?) = Unit
+
+  override fun read(json: Json, jsonData: JsonValue, type: Class<*>?): T
+}
+
+/**
+ * Factory function to create to create a read-only serializer
+ */
+inline fun <reified T> readOnlySerializer(crossinline reader: (Json, JsonValue, Class<*>?) -> T): Json.Serializer<T> =
+    object : KtxReadOnlySerializer<T> {
+      override fun read(json: Json, jsonData: JsonValue, type: Class<*>?): T = reader(json, jsonData, type)
+    }
+
+/**
+ * Factory function to create a read-only serializer.
+ * The same as [readOnlySerializer] but the lambda ignores [Class<*>][Class] parameter.
+ */
+inline fun <reified T> readOnlySerializer(crossinline read: (Json, JsonValue) -> T): Json.Serializer<T> =
```

Comment:
I don't think `T` has to be reified.

Reply:
Same goes for other functions.

Reply:
I was playing with different options a bit and just forgot to remove this one. My bad.